src/main/java/com/shopify/store/Application.java

```diff
@@ -0,0 +1,57 @@
+package com.shopify.store;
+
+import java.math.BigDecimal;
+import java.util.ArrayList;
+
+/**
+ *  Application here where one can enter and add in product, we can also input an ENUM check in the future to look at categories of products
+ */
+public class Application {
+    public static BigDecimal cart (ArrayList<FruitProduct> cartList) {
+        var discountZero = new Discount(new DiscountZeroStrategy());
+        var discountBOGO = new Discount(new DiscountBOGOStrategy());
+        var discountPercent = new Discount(new DiscountPercentageStrategy());
+       BigDecimal total = new BigDecimal(0);
+       for (int i = 0; i < cartList.size(); i++) {
```

Comment:
Use `for-each` instead: `for (var product : cartList)`

Reply:
Good reminder, thanks. I tried to use cartList.forEach((product) -> { total+= etc}); prior but with the BigDecimal i was having issues and took it out. I'll change it to lambda

Reply:
Java can't capture local variables (like `total`) mutably. You could do this with nasty hacks like 1 element arrays and shit https://stackoverflow.com/questions/51766676/how-to-avoid-defining-single-element-array-in-java-lambda-expression

But for summing more generally, you can use [`IntStream.sum()`](https://docs.oracle.com/javase/8/docs/api/java/util/stream/IntStream.html). `DoubleStream.sum()` also exists. But in your case, because there's no special methods for `Stream<BigDecimal>. Instead, you can use [`Stream.reduce()`](https://docs.oracle.com/javase/tutorial/collections/streams/reduction.html). In this case, `yourBigDecimalStream.reduce(BigDecimal.ZERO, BigDecimal::add);`